webapp/vue.md

```diff
@@ -0,0 +1,53 @@
+# Vue – Code Guidelines
+
+## We use single-file components
+
+Each component lives in a single file, containing:
+- its `template` (the DOM structure)
+- its `script` (including `props`, `data` and `methods` among other things)
+- its `style` (defining the look of the component)
+
+See the [Vue.js docs](https://vuejs.org/v2/guide/single-file-components.html) for more details.
+
+Placed in the same folder are also:
+- the test file (e.g. `MyComponent.spec.js`)
+- the storybook file (e.g. `MyComponent.story.js`)
+
+## We use typed props
+
+Vue.js allows us to define component props either as strings or as objects with `type`, `default` and `required` values. Always go for the second option!
+
+Also: define defaults _only and always_ for non-required props.
```

Comment:
I feel like we did the opposite in our pairing session the other day, oder?
we removed the `required: true`, and gave it `default: 'alert'`, no?

Reply:
Yep, that's what I was trying to say in this sentence. :wink: I'll rephrase it!